processing/src/main/java/org/apache/carbondata/processing/newflow/sort/impl/ParallelReadMergeSorterImpl.java

```diff
@@ -102,21 +102,14 @@ public void initialize(SortParameters sortParameters) {
     }
     this.executorService = Executors.newFixedThreadPool(iterators.length);
 
-    // First prepare the data for sort.
-    Iterator<CarbonRowBatch>[] sortPrepIterators = new Iterator[iterators.length];
-    for (int i = 0; i < sortPrepIterators.length; i++) {
-      sortPrepIterators[i] = new SortPreparatorIterator(iterators[i], inputDataFields);
-    }
-
-    for (int i = 0; i < sortDataRows.length; i++) {
-      executorService
-          .submit(new SortIteratorThread(sortPrepIterators[i], sortDataRows[i], sortParameters));
-    }
-
     try {
+      for (int i = 0; i < sortDataRows.length; i++) {
+        executorService
+            .submit(new SortIteratorThread(iterators[i], sortDataRows[i], sortParameters));
```

Comment:
should not break the line at function call, move `submit` to previous line and bread at parameters